x-pack/plugins/maps/public/store/store.js

```diff
@@ -20,17 +18,12 @@ const enhancers = [ applyMiddleware(thunk) ];
 window.__REDUX_DEVTOOLS_EXTENSION
   && enhancers.push(window.__REDUX_DEVTOOLS_EXTENSION__());
 
-export const getStore = function () {
-  if (store) {
```

Comment:
Rather than create a new store for each new map, why not just reset the state using a reducer as described here and then keep the singleton store: https://stackoverflow.com/questions/35622588/how-to-reset-the-state-of-a-redux-store

I tested this locally by:

1.  Creating an action to be dispatched on new or saved map load in the map controller
2. Updating the code here, mostly just adapted from my link above:
```
let store;

const appReducer = combineReducers({
  map,
  ui
});

const rootReducer = (state, action) => {
  if (action.type === LOAD_SAVED_OR_NEW_MAP) {
    state = undefined;
  }

  return appReducer(state, action);
};

const enhancers = [ applyMiddleware(thunk) ];
window.__REDUX_DEVTOOLS_EXTENSION
  && enhancers.push(window.__REDUX_DEVTOOLS_EXTENSION__());

export function createMapStore() {
  if (store) {
    return store;
  }
...
```
Thoughts?

Reply:
> why not just reset the state using a root reducer as described here and then keep the singleton store

Because we will need multiple instances of the store. If you reset the store for one embeddable, that would reset the store for the other embeddables if they share a store. Each embeddable will need its own store that is independent from the others.

Reply:
Fair enough. I've looked into this a bit and played around with it. The alternative idea we discussed offline was the possibility of creating slices of identical store keyed with a hash representing each map instance. You asked if it was possible to isolate actions to affect only their respective chunks of state. The answer is natively no, with a little code work maybe, with a library, *yes (but not how I'd like).

The resulting reducer from the `combineReducers` function out of the box will share actions with all reducers. There is [the option to delegate](https://redux.js.org/recipes/structuring-reducers/beyond-combinereducers#sharing-data-between-slice-reducers) to pieces of the store based upon a given action, and while this is really interesting and provides great options for optionally including pieces of other stores, it doesn't fit our use case as it delegates based on actions rather than store chunks. The [multireducer](https://github.com/erikras/multireducer) library handles the scenario I've described but not in an as dynamic or hands-off way as I'd prefer.

I do think we'd reap some nice benefits in using the same store like potential to share state when convenient and compat with redux dev tools but I don't think that should necessarily hold up this PR. The technique you're using is inline with the redux [recommendations for complete instance isolation](https://redux.js.org/recipes/isolating-redux-sub-apps) which should work fine for now.